packages/amplify-category-auth/resources/cloudformation-templates/auth-template-force-alias.yml.ejs

```diff
@@ -0,0 +1,1240 @@
+      <% } %>
+      <% if (props.aliasAttributes && props.aliasAttributes.length > 0) { %>
+      AliasAttributes: !Ref aliasAttributes
+      <% } %>
+      MfaConfiguration: !Ref mfaConfiguration
+      SmsVerificationMessage: !Ref smsVerificationMessage
+      SmsAuthenticationMessage: !Ref smsAuthenticationMessage
+      SmsConfiguration:
+        SnsCallerArn: !GetAtt SNSRole.Arn
+        ExternalId: <%=`${props.resourceNameTruncated}_role_external_id`%>
+    <%if (props.mfaConfiguration != 'OFF') { %>
+    DependsOn: SNSRole
+    <% } %>
+  <%if (!props.breakCircularDependency && props.triggers && props.dependsOn) { %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('CreateAuthChallenge'))) { %>
+  UserPoolCreateAuthChallengeLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>CreateAuthChallengeName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('CustomMessage'))) { %>
+  UserPoolCustomMessageLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>CustomMessageName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('DefineAuthChallenge'))) { %>
+  UserPoolDefineAuthChallengeLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>DefineAuthChallengeName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PostAuthentication'))) { %>
+  UserPoolPostAuthenticationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PostAuthenticationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PostConfirmation'))) { %>
+  UserPoolPostConfirmationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PostConfirmationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PreAuthentication'))) { %>
+  UserPoolPreAuthenticationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
```

Comment:
EJS question, should UserPool be in quotes here ? as in
"DependsOn" : [ "UserPool" ]

Reply:
More of a yml question than ejs, but yes the syntax here is what we want, since UserPool refers to the Cognito UserPool defined in the above yml